vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -409,28 +450,16 @@ void clientDeselectAll() {
             // ignore event if the checkBox was meant to be hidden
             return;
         }
-        doUpdateSelection(Collections.emptySet(), getSelectedItems(), true);
+        doUpdateSelection(Collections.emptyMap(), mapItemsById(getSelectedItems()), true);
         selectionColumn.setSelectAllCheckboxState(false);
         selectionColumn.setSelectAllCheckboxIndeterminateState(false);
     }
 
-    private void doUpdateSelection(Set<T> addedItems, Set<T> removedItems,
-            boolean userOriginated) {
-        Map<Object, T> addedItemsMap = mapItemsById(addedItems);
-        Map<Object, T> removedItemsMap = mapItemsById(removedItems);
-        addedItemsMap.keySet().stream().filter(removedItemsMap::containsKey)
-                .collect(Collectors.toList()).forEach(key -> {
-                    addedItemsMap.remove(key);
-                    removedItemsMap.remove(key);
-                });
-        doUpdateSelection(addedItemsMap, removedItemsMap, userOriginated);
-    }
-
     private void doUpdateSelection(Map<Object, T> addedItems,
             Map<Object, T> removedItems, boolean userOriginated) {
-
-        if (selected.keySet().containsAll(addedItems.keySet()) && Collections
-                .disjoint(selected.keySet(), removedItems.keySet())) {
+        Set<Object> selectedIds = getSelectedItemIds();
```

Comment:
This could be refactored (method removed) in future by someone in the team. It's the only place where `getSelectedItemIds` is used (apart from the tests).
I will discuss with the team if we could proceed with enhancing the documentation for this particular method or consider a public API.

Reply:
I guess from this point I will overtake it and add needed comments to the documentation.